hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsencePeriodTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRAbsence_BAO_HRAbsencePeriod as AbsencePeriod;
+use CRM_HRAbsence_Test_Fabricator_HRAbsencePeriod as AbsencePeriodFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsencePeriodTest
+ *
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsencePeriodTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface, TransactionalInterface {
+
+  /**
+   * @var array
+   *  An array of AbsencePeriods created by the setUp() method
+   */
+  private $absencePeriods = [];
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()->installMe(__DIR__)->apply();
+  }
+
+  public function setUp() {
+    $absencePeriodTable = AbsencePeriod::getTableName();
+    // We need to disable fk checks in order to avoid constraint
+    // errors when truncating the absence period table
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 0;");
+    CRM_Core_DAO::executeQuery("TRUNCATE $absencePeriodTable");
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 1;");
+
+    $this->absencePeriods[] = AbsencePeriodFabricator::fabricate([
+      'name' => 'Period 1',
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
```

Comment:
You mean the create method of the BAO? If yes, that is not a good idea because of this https://gist.github.com/davialexandre/36c27089b2dbb8630fa4ba1a1d942c4f